src/controllers/dynakube/dtclient_reconciler.go

```diff
@@ -202,9 +210,28 @@ func (r *DynatraceClientReconciler) Reconcile(ctx context.Context, instance *dyn
 		})
 	}
 
+	if updateSecret {
+		if err := r.Client.Update(ctx, secret); err != nil {
+			return dtc, updateCR, fmt.Errorf("failed to update secret: %s", err)
+		}
+	}
+
 	return dtc, updateCR, nil
 }
 
+func (r *DynatraceClientReconciler) getTokensFromSecret(ctx context.Context, instance *dynatracev1beta1.DynaKube) (*corev1.Secret, error) {
```

Comment:
this doesn't only get the tokens from the secret
maybe splitting this into loading the secret and then assigning the tokens in a separate stuff would improve readability here